src/thunks/searchThunk.js

```diff
@@ -4,51 +4,49 @@ const treeherderBaseURL = 'https://treeherder.mozilla.org';
 
 export const fetchRecentRevisions = createAsyncThunk(
   'search/fetchRecentRevisions',
-  async (repository, { rejectWithValue }) => {
-    let response;
-    try {
-      response = await fetch(
-        `${treeherderBaseURL}/api/project/${repository}/push/`,
-      )
-        .then((res) => res.json())
-        .then((res) => res.results);
-    } catch (err) {
-      return rejectWithValue(err.response.data);
+  async (repository, { getState, requestId }) => {
+    const { currentRequestID, loading } = getState().search;
+    if (loading !== 'pending' || requestId !== currentRequestID) {
+      return [];
     }
+    const response = await fetch(
+      `${treeherderBaseURL}/api/project/${repository}/push/`,
+    )
+      .then((res) => res.json())
+      .then((res) => res.results);
```

Comment:
Style comment:
I'd wait this more like this:
```js
const res = await fetch(...)
const json = await res.json();
return res.results;
```

Indeed in a code such as this one, I usually find it more readable to linearize it this way. This is fully equivalent so it's really a style preference.

It may still be necessary to use the Promise API directly when running concurrent tasks, but that's not that common.

Reply:
I'm guessing you meant (in the last line) `return json.results;`?

Reply:
Yes! oops :-)